From 2e6e9054662912d3dea4e038673304e115b8ab6d Mon Sep 17 00:00:00 2001
From: Josh Gao <jmgao@google.com>
Date: Wed, 28 Oct 2020 13:56:23 -0700
Subject: [PATCH] Let adbd set service.adb.tcp.port.

Commit 67c36884 changed the label of service.adb.tcp.port to allow
vendor init to set it, but accidentally prevented adbd from setting it,
which broke `adb tcpip`.

Bug: http://b/171280882
Test: `adb tcpip`
Change-Id: Ifeeda5c4f06451158fc7e43ca23f580092008fe7
(cherry picked from commit 0cac6fd17a341cc64e9aca2e9b402234b08264d8)
---
 prebuilts/api/30.0/private/adbd.te        | 3 ++-
 prebuilts/api/30.0/private/vendor_init.te | 3 +++
 prebuilts/api/30.0/public/property.te     | 2 +-
 private/adbd.te                           | 3 ++-
 private/vendor_init.te                    | 3 +++
 public/property.te                        | 2 +-
 6 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/prebuilts/api/30.0/private/adbd.te b/prebuilts/api/30.0/private/adbd.te
index b84a86607..5531ad41f 100644
--- a/prebuilts/api/30.0/private/adbd.te
+++ b/prebuilts/api/30.0/private/adbd.te
@@ -87,8 +87,9 @@ set_prop(adbd, powerctl_prop)
 set_prop(adbd, ffs_prop)
 set_prop(adbd, exported_ffs_prop)
 
-# Set service.adb.tls.port, persist.adb.wifi. properties
+# Set service.adb.tcp.port, service.adb.tls.port, persist.adb.wifi.* properties
 set_prop(adbd, adbd_prop)
+set_prop(adbd, adbd_config_prop)
 
 # Access device logging gating property
 get_prop(adbd, device_logging_prop)
diff --git a/prebuilts/api/30.0/private/vendor_init.te b/prebuilts/api/30.0/private/vendor_init.te
index 6a68f1fed..83f001d6a 100644
--- a/prebuilts/api/30.0/private/vendor_init.te
+++ b/prebuilts/api/30.0/private/vendor_init.te
@@ -5,3 +5,6 @@ dontaudit vendor_init sysfs:dir write;
 
 # TODO(b/140259336) We want to remove vendor_init in the long term but allow for now
 allow vendor_init system_data_root_file:dir rw_dir_perms;
+
+# Let vendor_init set service.adb.tcp.port.
+set_prop(vendor_init, adbd_config_prop)
diff --git a/prebuilts/api/30.0/public/property.te b/prebuilts/api/30.0/public/property.te
index 26cf6c3a7..43b09db8d 100644
--- a/prebuilts/api/30.0/public/property.te
+++ b/prebuilts/api/30.0/public/property.te
@@ -113,7 +113,6 @@ compatible_property_only(`
 ')
 
 # Properties which can be written only by vendor_init
-system_vendor_config_prop(adbd_config_prop)
 system_vendor_config_prop(apk_verity_prop)
 system_vendor_config_prop(cpu_variant_prop)
 system_vendor_config_prop(exported_audio_prop)
@@ -133,6 +132,7 @@ system_vendor_config_prop(vndk_prop)
 system_vendor_config_prop(virtual_ab_prop)
 
 # Properties with no restrictions
+system_public_prop(adbd_config_prop)
 system_public_prop(audio_prop)
 system_public_prop(bluetooth_a2dp_offload_prop)
 system_public_prop(bluetooth_audio_hal_prop)
diff --git a/private/adbd.te b/private/adbd.te
index b84a86607..5531ad41f 100644
--- a/private/adbd.te
+++ b/private/adbd.te
@@ -87,8 +87,9 @@ set_prop(adbd, powerctl_prop)
 set_prop(adbd, ffs_prop)
 set_prop(adbd, exported_ffs_prop)
 
-# Set service.adb.tls.port, persist.adb.wifi. properties
+# Set service.adb.tcp.port, service.adb.tls.port, persist.adb.wifi.* properties
 set_prop(adbd, adbd_prop)
+set_prop(adbd, adbd_config_prop)
 
 # Access device logging gating property
 get_prop(adbd, device_logging_prop)
diff --git a/private/vendor_init.te b/private/vendor_init.te
index 6a68f1fed..83f001d6a 100644
--- a/private/vendor_init.te
+++ b/private/vendor_init.te
@@ -5,3 +5,6 @@ dontaudit vendor_init sysfs:dir write;
 
 # TODO(b/140259336) We want to remove vendor_init in the long term but allow for now
 allow vendor_init system_data_root_file:dir rw_dir_perms;
+
+# Let vendor_init set service.adb.tcp.port.
+set_prop(vendor_init, adbd_config_prop)
diff --git a/public/property.te b/public/property.te
index 26cf6c3a7..43b09db8d 100644
--- a/public/property.te
+++ b/public/property.te
@@ -113,7 +113,6 @@ compatible_property_only(`
 ')
 
 # Properties which can be written only by vendor_init
-system_vendor_config_prop(adbd_config_prop)
 system_vendor_config_prop(apk_verity_prop)
 system_vendor_config_prop(cpu_variant_prop)
 system_vendor_config_prop(exported_audio_prop)
@@ -133,6 +132,7 @@ system_vendor_config_prop(vndk_prop)
 system_vendor_config_prop(virtual_ab_prop)
 
 # Properties with no restrictions
+system_public_prop(adbd_config_prop)
 system_public_prop(audio_prop)
 system_public_prop(bluetooth_a2dp_offload_prop)
 system_public_prop(bluetooth_audio_hal_prop)
-- 
2.17.1

